python/tvm/relay/op/contrib/dnnl.py

```diff
@@ -353,6 +370,7 @@ def alter_conv(attrs, inputs, tinfos, out_type):
     paddings = ",".join([str(x) for x in attrs.get_int_tuple("padding")])
     strides = ",".join([str(x) for x in attrs.get_int_tuple("strides")])
     dilates = ",".join([str(x) for x in attrs.get_int_tuple("dilation")])
+    dtype = get_dtype(weight)
```

Comment:
get dtype from data by convention?

Reply:
Here the `data` is a `CallNode` which prohibit the query for dtype, except the first input layer. This is also the reason why we couldn't deal with the case with different dtype for inputs, weights or dst.

Reply:
I'm wondering if TVM has something like infer_data_type pass (which propagate the datatypes from the very beginning to the end) or not?

Reply:
Good point! TVM has a pass named `InferType`, I will try to apply this pass before querying the dtype.

Reply:
I tried to run the `InferType` pass before `AlterOpLayout`, but it still raise the error:
> The type checker has not populated the checked_type for this node.

Reply:
We might need to check why the infertype pass doesn't update the info to the `data`, but I think you can submit the PR with `weight` at this moment.

Reply:
OK